Source/player.cpp

```diff
@@ -3577,7 +3578,19 @@ void CheckPlrSpell()
 
 	if (!addflag) {
 		if (myPlayer._pRSplType == RSPLTYPE_SPELL) {
-			myPlayer.Say(HeroSpeech::NotEnoughMana);
+			HeroSpeech sentence;
+			switch (spellcheck) {
+			case SpellCheckResult::Fail_NoMana:
+				sentence = HeroSpeech::NotEnoughMana;
+				break;
+			case SpellCheckResult::Fail_Level0:
+				sentence = HeroSpeech::ICantCastThatYet;
+				break;
+			default:
+				sentence = HeroSpeech::ICantDoThat;
+				break;
+			}
+			myPlayer.Say(sentence);

```

Comment:
I don't think we are actually saving anything by using the var and it compiles to the same code:
```suggestion
			switch (spellcheck) {
			case SpellCheckResult::Fail_NoMana:
				myPlayer.Say(HeroSpeech::NotEnoughMana);
				break;
			case SpellCheckResult::Fail_Level0:
				myPlayer.Say(HeroSpeech::ICantCastThatYet);
				break;
			default:
				myPlayer.Say(HeroSpeech::ICantDoThat);
				break;
			}
```